### DISCUSSION TITLE
If there are running jobs in the Volcano queue, is it permissible to directly delete the queue?

### DISCUSSION THREAD

**Discussion by hzp01** [fabric8io/kubernetes-client#5658] [Q&A] [answered]
volcanoClient.queues().withName(queueName).delete();

**Comment by rohanKanojia** [comment]
Sorry, but could you please elaborate your question?

**Reply by hzp01** [reply]
I performed the operation to delete a queue, forgetting that there were still running jobs under the queue. The queue was deleted successfully without affecting the jobs. Should I manually query the status of jobs under the queue before executing the deletion in my code?

**Reply by rohanKanojia** [reply]
Do you have some [finalizers](https://kubernetes.io/blog/2021/05/14/using-finalizers-to-control-deletion/) added in Queue to check for this case?

**Reply by hzp01** [reply]
The following is the logical check I implemented in my code.
```
        NamespacedVolcanoClient volcanoClient = VolcanoClient.getVolcanoClient();
        GenericKubernetesResourceList list = VolcanoJobQuery.volcanoJobQueryList();
        Boolean haveNotEndJobInQueue = false;
        List<String> jobEndStatus = Arrays.asList("succeeded", "failed", "unknown", "completed", "failed");
        if (list != null && !list.getItems().isEmpty()) {
            haveNotEndJobInQueue = list.getItems().stream().filter(item -> {
                try {
                    return queueName.equals(((LinkedHashMap) item.get("spec")).get("queue").toString());
                } catch (Exception e) {
                    return false;
                }
            }).anyMatch(item -> {
                try {
                    return jobEndStatus.contains(((LinkedHashMap) ((LinkedHashMap) item.get("status")).get("state")).get("phase").toString().toLowerCase());
                } catch (Exception e) {
                    System.out.println(e);
                    return false;
                }
            });
        }
        if(!haveNotEndJobInQueue){
            volcanoClient.queues().withName(queueName).delete();
        }
        volcanoClient.close();
```

**Reply by rohanKanojia** [reply]
I see. What problem are you facing with this code?

**Reply by hzp01** [reply]
The code is lengthy, and I'm wondering if there is already an API that implements this functionality, but I'm not sure.

**Reply by rohanKanojia** [reply, marked as answer]
I'm not aware of any API in fabric8 k8s client that implements this. You would need to check with volcano community on how they handle this Queue delete precondition. I checked their Queue controller but couldn't find any implemented finalizer. Not 100% sure maybe this can be added as a finalizer in Volcano?

**Reply by hzp01** [reply]
Okay, thank you.
